phac-nml/irida-next
# Upgrade: Updated modal to have action buttons and JavaScript open

## What does this PR do and why?
_Describe in detail what your merge request does and why._

**Renamed modal to dialog**: This provides better naming consistency as it is based of the native _dialog_ element

1. Updated to use native [browser dialog element
](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog)
-  Allows browser to handle accessibility
- Compatible with all major browser
2. Added **Header** component to allow for showing or hiding the title.
3. Added _action buttons_ allowing for primary and secondary actions
4. Allow the modal to be opened and closed by JavaScript
5. Cleaned up all styles


## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

### Light with action buttons
![image](https://github.com/phac-nml/irida-next/assets/11295750/0d842da1-ab14-4bb9-a44c-a1cce2c3b182)

### Dark with action buttons
![image](https://github.com/phac-nml/irida-next/assets/11295750/4bb8d50e-1e02-4c01-b00f-2aede85b0db5)

### Multiple Sections
![image](https://github.com/phac-nml/irida-next/assets/11295750/57c57eda-feb7-4a3b-a47c-fe5925da0f9f)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

View all styles in LookBook.  Make sure to check the dark modes as well:

1. [Default modal](http://localhost:3000/rails/lookbook/inspect/viral_dialog/default): Title, content, and no buttons (besides close).
2. [Small](http://localhost:3000/rails/lookbook/inspect/viral_dialog/small): same as default, only slightly narrower
3. [Large](http://localhost:3000/rails/lookbook/inspect/viral_dialog/large): Same as default, except wider
4. [Extra Large](http://localhost:3000/rails/lookbook/inspect/viral_dialog/extra_large): Same as default, except fills majority of page width
5. [With Primary Buttons](http://localhost:3000/rails/lookbook/inspect/viral_dialog/with_primary_action): Title, content and action buttons
6. [With multiple section](http://localhost:3000/rails/lookbook/inspect/viral_dialog/with_multiple_sections): Same as default except has multiple sections of content separated by a horizontal rul

## PR acceptance checklist

This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
